src/parser/selectorMatches.ts

```diff
@@ -14,10 +12,10 @@ export function selectorMatches(element, selector) {
     matcher = new RegExp('#' + id + '(?![a-zA-Z\\-]+)', 'i');
     selector = selector.replace(matcher, '');
   }
-  if (classNames && selector.length) {
-    classNames = classNames.split(' ');
-    for (i = classNames.length; i--; ) {
-      matcher = new RegExp('\\.' + classNames[i] + '(?![a-zA-Z\\-]+)', 'i');
+  if (classNames != null && selector.length) {
```

Comment:
why did you add this?
I think you should revert this

Reply:
<img width="1093" alt="image" src="https://user-images.githubusercontent.com/91561865/220834731-3cb2494c-71bf-4f9b-b4b8-8e2cc74b4fdd.png">


This is the method signature `Element.getAttribute(qualifiedName: string): string | null `

So I use the `splitClassNames` variable to store

This method will only have two return values `string` or `null`

Reply:
@ShaMan123  Can you re-examine it for me?  Thank you so much

Reply:
I understand. Good research.
I prefer we leave it as is since `if(null)` will never run and this way we only change lint stuff
Also we use strict equality checks `!==`

Reply:
```suggestion
  if (classNames && selector.length) {
```

Reply:
ok, I understand,